dynamic_config/dynamic_config_manager.py

```diff
@@ -192,19 +193,156 @@ def serialize(self) -> str:
     return self._active_config.SerializeToString()
 
 
+class DynamicEndpointsConfigManager(DynamicConfigManager):
+  """Encapsulates dynamic endpoint configuration."""
+
+  def __init__(self, config: endpoints_config_manager_pb2.DynamicEndpointsConfigManagerSettings):
+    """Create a DynamicEndpointsConfigManager."""
+    DynamicConfigManager.__init__(self)
+    self._active_config = discovery_pb2.DiscoveryResponse()
+    self._active_config.version_info = "1"
+    self._active_config.resources.add()
+
+    # The endpoints to publish
+    self._cluster_load_assignment = endpoint_pb2.ClusterLoadAssignment()
+    self._cluster_load_assignment.cluster_name = config.cluster.name
+    self._cluster_load_assignment.endpoints.add()
+
+    # Convience hook to add endpoints.
+    self._lb_endpoints = self._cluster_load_assignment.endpoints[0].lb_endpoints
+
+    endpoints = self._parseAvailableEndpoints(config.cluster.endpoints)
+    self._inactive_endpoints = []
+    self._randomlyAssignInitialEndpoints(endpoints)
+    # TODO(kbaichoo): refactor into a shared constants file.
+    self._output_file = config.output_file if config.output_file else "new_eds.pb"
+    # TODO(kbaichoo): Handle nanos get ignored
+    self._refresh_interval = config.refresh_interval.ToSeconds()
+    # Tracks the last mutated action. Used for testing.
+    self._last_mutate_action = None
+
+  def _parseAvailableEndpoints(
```

Comment:
This method doesn't seem to be using the class instance (self). Can we make it into a standalone function instead to simplify the class?

Possibly applies to multiple locations.

Reply:
done.